engine/src/flutter/display_list/geometry/dl_path_unittests.cc

```diff
@@ -553,6 +553,405 @@ TEST(DisplayListPath, ConstructFromImpellerEqualsConstructFromSkia) {
   EXPECT_EQ(DlPath(path_builder, DlPathFillType::kNonZero), DlPath(sk_path));
 }
 
+namespace {
+class TestPathElement {
+ public:
+  enum class Type {
+    kMoveTo,
+    kLineTo,
+    kQuadTo,
+    kConicTo,
+    kCubicTo,
+    kClose,
+  };
+
+  static TestPathElement MoveTo(const DlPoint& p) {
+    return {
+        .type = Type::kMoveTo,
+        .p1 = p,
+    };
+  }
+  static TestPathElement LineTo(const DlPoint& p) {
+    return {
+        .type = Type::kLineTo,
+        .p1 = p,
+    };
+  }
+  static TestPathElement QuadTo(const DlPoint& cp, const DlPoint& p2) {
+    return {
+        .type = Type::kQuadTo,
+        .p1 = cp,
+        .p2 = p2,
+    };
+  }
+  static TestPathElement ConicTo(const DlPoint& cp,
+                                 const DlPoint& p2,
+                                 DlScalar w) {
+    return {
+        .type = Type::kConicTo,
+        .p1 = cp,
+        .p2 = p2,
+        .weight = w,
+    };
+  }
+  static TestPathElement CubicTo(const DlPoint& cp1,
+                                 const DlPoint& cp2,
+                                 const DlPoint& p2) {
+    return {
+        .type = Type::kCubicTo,
+        .p1 = cp1,
+        .p2 = cp2,
+        .p3 = p2,
+    };
+  }
+  static TestPathElement Close() { return {.type = Type::kClose}; }
+
+  const Type type;
+  const DlPoint p1;
+  const DlPoint p2;
+  const DlPoint p3;
+  const DlScalar weight;
+};
+
+class TestPathReceiver : public DlPathReceiver {
```

Comment:
no action required: This class should just be a gmock.

Reply:
I took a quick look at gmock just now, but it seems focused on just asserting that methods are called whereas this class is checking the data they are called with.

Reply:
There are hundreds of examples of using gmock to check the arguments of function calls in our codebase.  Here is one example: https://github.com/flutter/flutter/blob/b16430b2fd57a5aa6b6c680cdbda5582506fe120/engine/src/flutter/shell/platform/android/external_view_embedder/external_view_embedder_unittests.cc#L288

This is asserting that `AcquireFrame` with arguments that equal `frame_size`.

Reply:
I don't understand the formatting used there, it looks like it is placing expectations on return values, but my methods are all void and I'm placing expectations on the data they receive. Again, maybe that's in there, but I'd need to study gmock further - likely worth the work.

Reply:
Maybe this one is easier to see: https://github.com/flutter/flutter/blob/b16430b2fd57a5aa6b6c680cdbda5582506fe120/engine/src/flutter/impeller/renderer/backend/gles/unique_handle_gles_unittests.cc#L29

This is asserting that GenTextures is being called with 2 arguments, the first one must be 1 the second can be anything ("_" is a wildcard matcher).

I think this would be a good opportunity to learn how to use gmock.  It's a very useful skill and I'm happy to help.

Reply:
TIL-gmock. The tests are slightly more wordy, but the intent is outlined in the test itself and the mock receiver class was 100x simpler than the tester harness class.

Reply:
See also https://github.com/flutter/flutter/issues/165336